code/missionui/redalert.cpp

```diff
@@ -979,9 +985,15 @@ void red_alert_bash_wingman_status()
 						}
 					}
 
+					float max_hull;
+					if (pobjp->special_hitpoints)
+						max_hull = pobjp->ship_max_hull_strength;
+					else
+						max_hull = Ship_info[pobjp->ship_class].max_hull_strength;
+
 					// restore hull (but not shields)
-					if (ras->hull >= 0.0f && ras->hull <= (pobjp->initial_hull * pobjp->ship_max_hull_strength / 100.0f))
```

Comment:
I wonder what was up with this complicated formula.

Reply:
For parse objects, the initial hull is stored as a number between 0 and 100.  For ships that are present, the hull is stored as the actual number of hitpoints.  This converts from the first to the second.